# HG changeset patch
# User Petr Sumbera <petr.sumbera@oracle.com>
# Date 1725269259 25200
#      Mon Sep 02 02:27:39 2024 -0700
# Node ID 1e24dcd1c214c7dc6342c5f9f246d68655fe3771
# Parent  99919665f083c45e705d4feeec7266f2c6474ae8
Workaround for Bug 1669739 - webrender (_stack_*)

Solaris libc doesn't provide __stack_chk_*; GCC links -lssp automatically,
but rustc does not. Explicitly add -lssp for stack-protected C/C++ code
compiled via Rust build scripts.

--- firefox-153.0/build/moz.configure/toolchain.configure
+++ firefox-153.0/build/moz.configure/toolchain.configure
@@ -2717,6 +2717,12 @@
             flags.append("-fstack-protector-strong")
             ldflags.append("-fstack-protector-strong")
 
+            # Solaris libc doesn't provide __stack_chk_*; GCC links -lssp automatically,
+            # but rustc does not. Explicitly add -lssp to avoid linker errors when using
+            # -fstack-protector in C/C++ code compiled via Rust build scripts.
+            if target.os == "SunOS":
+                ldflags.append("-lssp")
+
             if (
                 c_compiler.type == "clang"
                 and target.os not in ("WINNT", "OSX", "OpenBSD", "iOS")
